<launch>
    <arg name='port' default='/dev/ttyUSB0' />
    <arg name='baud' default='57600' />

    <group ns='arduino'>
        <node pkg='rosserial_python' type='ardu_serial.py' name='node_serial' output='screen'>
            <param name='port' value='$(arg port)'/>
            <param name='baud' value='$(arg baud)'/>
        </node>
    </group>

    <arg name='duration' default='5' />

    <group ns='nuc'>
        <param name='duration' value='$(arg duration)'/>

        <node pkg='robotic_sas_auv_ros' type='node_accumulator.py' name='node_accumulator' output='screen'/>
        <node pkg='robotic_sas_auv_ros' type='node_guidance.py' name='node_guidance' output='screen'/>
        <node pkg='robotic_sas_auv_ros' type='node_navigation.py' name='node_navigation' output='screen'/>
        <node pkg='robotic_sas_auv_ros' type='node_control.py' name='node_control' output='screen'/>
    </group>
</launch>